app/services/affiliate-slots.js

```diff
@@ -175,16 +204,41 @@ export default Service.extend({
         resolve(undefined);
       }
 
-      // get the units that fulfill the targeting on search
-      const targeting = this._getTargetingOnSearch(query);
-      const availableUnits = this._getUnitsWithTargeting(targeting)
-        // filter type of ad - isBig can be undefined, let's convert both to boolean
-        .filter(u => !!u.isBig === !!isBig)
-        // filter units disabled on search page
-        .filter(u => !u.disableOnSearch);
-
-      // fetch only the first unit if available
-      resolve(availableUnits.length > 0 ? availableUnits[0] : undefined);
+      if (isBig) {
+        // get the units that fulfill the targeting on search
+        const targeting = this._getTargetingOnSearch(query);
+        const availableUnits = this._getUnitsWithTargeting(targeting)
+          // we only want big units at this point
+          .filter(u => !!u.isBig)
+          // filter units disabled on search page
+          .filter(u => !u.disableOnSearch);
+
+        // fetch only the first unit if available
+        resolve(availableUnits.length > 0 ? availableUnits[0] : undefined);
+      } else {
+        const url = this.fetch.getServiceUrl('knowledge-graph', `/affiliates/${this.currentWikiId}`);
+
+        this.fetch.fetchAndParseResponse(url, {}, AffiliatesFetchError)
+          .then((response) => {
+            const targeting = flattenKnowledgeGraphTargeting(response);
+
+            // get the units that fulfill the campaign and category
+            const availableUnits = this._getUnitsWithTargeting(targeting)
+              // we only want only small at this point
+              .filter(u => !u.isBig)
+              // filter units disabled on article page
+              .filter(u => !u.disableOnSearch);
+
+            // fetch only the first unit if available
+            resolve(availableUnits.length > 0 ? availableUnits[0] : undefined);
+          })
+          .catch((error) => {
+            // log and do not raise anything
+            console.error(error);
```

Comment:
I think the linter might object to `console` statements.

Reply:
d3f3720